src/networking.c

```diff
@@ -40,8 +40,23 @@
 #include <sys/uio.h>
 #include <math.h>
 #include <ctype.h>
+#include <stdbool.h>
 #include <stdatomic.h>
 
+typedef enum {
+    CLIENT_REPLY_PAYLOAD_DATA = 0,
+    CLIENT_REPLY_PAYLOAD_BULK_OFFLOAD,
+} clientReplyPayloadType;
+
+/* Reply payload header */
+typedef struct __attribute__((__packed__)) payloadHeader {
```

Comment:
Why is this packed? I would generally prefer we let the compiler decide.

Reply:
Wanted headers to consume less bytes from client reply buffers. Removed it

Reply:
The struct is better packed to avoid wasting bytes with padding in the client's buffer. Since the buffer is accessed at arbitrary offsets anyway (not aligned), there's no benefit from compiler-added padding, so we might as well save the space.

Reply:
agree with @uriyage . will restore `__packed__` to make headers more frugal.